testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveTransactionalTable.java

```diff
@@ -1903,6 +1904,49 @@ public void testDeleteAfterMajorCompaction()
         });
     }
 
+    @Test
+    public void testInsertIntoUnbucketedTransactionalTableWithTaskWriterCountGreaterThan1()
+    {
+        withTemporaryTable("test_insert_into_unbucketed_transactional_table", true, false, NONE, tableName -> {
+            onTrino().executeQuery(format("CREATE TABLE %s  WITH (format='ORC', transactional=true) AS TABLE tpch.tiny.nation WITH NO DATA", tableName));
+            onTrino().executeQuery("SET SESSION task_writer_count = 2");
+            onTrino().executeQuery(format("INSERT INTO %s SELECT * FROM tpch.tiny.nation", tableName));
+            int numberOfCreatedFiles = onTrino().executeQuery(format("SELECT DISTINCT \"$path\" FROM %s", tableName)).getRowsCount();
+            assertEquals(numberOfCreatedFiles, 1, "There should be only 1 file created");
+            assertThat(onTrino().executeQuery(format("SELECT COUNT(*) from %s", tableName))).containsOnly(row(25));
+        });
+    }
+
+    @Test
+    public void testDataIsNotBrokenInUnbucketedTransactionalTableWithTaskWriterCountGreaterThan1()
```

Comment:
Do we need the test above? This one seems to cover same stuff + DELETE

Reply:
right, I will delete it